src/main/webapp/scripts/script.js

```diff
@@ -184,11 +178,24 @@ function getDomain() {
 
 function setLoadingOverlay() {
     var overlay = document.getElementsByClassName("overlay");
-    var overlayArray = Array.from(overlay);
     if (isLoading) {
-        overlayArray.map(elem => elem.classList.remove("hidden"))
+        showElements(overlay)
     } else {
-        overlayArray.map(elem => elem.classList.add("hidden"))
+        hideElements(overlay)
+    }
+}
+
+// Adds the class "hidden" to the array of elements
+function hideElements(elements) {
```

Comment:
For both hide and show Elements functions, if you added a `elem = document.getElementById(elemId)` line you could remove a lot of code since you will only have to pass a list of element ids into these methods